docker/docker-compose.development.yml

```diff
@@ -1,52 +1,69 @@
 version: '3.5'
 
-services:
+x-networked: &networked
+  networks:
+    - nhsd_bcnetwork
+
+x-depends-on-db: &dependsOnDb
+  <<: *networked
+  depends_on:
+    - db
+
+x-sa-password: &saPassword
+  - SA_PASSWORD: "MyTestPassword123!"
 
+services:
   db:
-    image: ${REGISTRY:-nhsd}/buying-catalogue-db:${TAG:-latest}
-    build: 
-      context: "../../DataModel"
-      dockerfile: Dockerfile
+    <<: *networked
+    image: mcr.microsoft.com/mssql/server:2019-latest
     container_name: nhsd_bcdb
     volumes:
       - nhsd_bcvolume:/var/opt/mssql/
-    networks:
-      - nhsd_bcnetwork
     environment:
-      - ACCEPT_EULA=Y
-      - SA_PASSWORD=MyTestPassword123!
-      - MSSQL_PID=Developer
-      - DB_NAME=buyingcatalogue
-      - DB_USER=NHSD
-      - DB_PASSWORD=DisruptTheMarket1!
+      <<: *saPassword
+      ACCEPT_EULA: "Y"
+    healthcheck:
+      test: /opt/mssql-tools/bin/sqlcmd -U sa -P $${SA_PASSWORD} -Q "SELECT 1;"
```

Comment:
@assoomro can you add your thoughts?

Reply:
I've had a similar conversation with @sam-roberts-bjss when I upgraded ORDAPIs database to mssql 2019. To make a longer story short, I agree with @sam-roberts-bjss that the checking the setup (entrypoint) script has finished is a good idea. 

Now, to answer Sam's question, this is, indeed, a way to replace the HEALTHCHECK statement in the datamodel's Dockerfile. Since we're no longer building a custom image for the db, this is away to achieve the same behaviour. 

I think it would be a good idea to modify the healthcheck to also perform a search for 'MSSQL CONFIG COMPLETE' (or whatever string we decide to output in our entrypoint.sh) in a log file.

Now the tricky bit is that you will not be able to use the same file paths as we used to do, due to the fact that Microsoft has made this image run as non-root user by default. I'd suggest having a look at the [Ordapi's database dockerfile](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/blob/development/src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Dockerfile) and [its configuration script](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/blob/development/src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/ShellScripts/configure-db.sh) as they are creating (and then checking) the .log file in a space where the non-root user has permissions.

Another alternative to solve the problems introduced by running as non-root user is to add 
`USER root` in the Dockerfile, however, I would highly suggest **NOT** to do that.

Reply:
Thanks @assoomro. I'll make the required changes.